ddl/cancel_test.go

```diff
@@ -195,10 +231,11 @@ func TestCancel(t *testing.T) {
 	hook := &ddl.TestDDLCallback{Do: dom}
 	i := 0
 	cancel := false
+	cancelWhenReorgNotStart := false
 
 	hookFunc := func(job *model.Job) {
 		if job.SchemaState == allTestCase[i].cancelState && !cancel {
-			if job.SchemaState == model.StateWriteReorganization && ddl.MayNeedReorg(job) && job.RowCount == 0 {
+			if !cancelWhenReorgNotStart && job.SchemaState == model.StateWriteReorganization && ddl.MayNeedReorg(job) && job.RowCount == 0 {
```

Comment:
`cancelWhenReorgNotStart` is always `true` after the first test case is run. Please check if this is expected.

Reply:
Good catch, fixed.